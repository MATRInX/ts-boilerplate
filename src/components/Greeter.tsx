import React, { SFC } from 'react';

export interface GreeterProps {
  name?: string
};

export const Greeter: SFC<GreeterProps> = (props) => {
  const { name } = props;
  return (
    <div>
      Hello {name} in world of typescript!
    </div>
  );
}

export default Greeter;