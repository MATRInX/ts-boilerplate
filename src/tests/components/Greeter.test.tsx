import React from 'react';
import { shallow } from 'enzyme';
import Greeter from '../../components/Greeter';

test('should correctly render Greeter', () => {
  const wrapper = shallow(<Greeter />);
  expect(wrapper).toMatchSnapshot();
});

test('should correctly render Greeter with provided name', () => {
  const name: string = 'Marcin';
  const wrapper = shallow(<Greeter name={name}/>);
  expect(wrapper).toMatchSnapshot();
});