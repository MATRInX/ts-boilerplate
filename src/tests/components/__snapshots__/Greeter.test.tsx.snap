// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render Greeter 1`] = `
<div>
  Hello 
   in world of typescript!
</div>
`;

exports[`should correctly render Greeter with provided name 1`] = `
<div>
  Hello 
  Marcin
   in world of typescript!
</div>
`;
